### PR TITLE
Dispatch events for Leads and Companies created by Pipedrive

### PR DESCRIPTION
**Pull request by alanhartless**
[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | y
| New feature? | 
| Automated tests included? | y
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

Contacts and companies created by Pipedrive's webhook (import) did not dispatch events so plugins and other listeners were not notified of changes made. This fixes that. 

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. Setup Pipedrive CRM webhook according to https://mautic.org/docs/en/plugins/pipedrive.html. 
2. Create a contact in Pipedrive
3. The contact should be created in Mautic
4. Go to the contact record and note that there are no entries under the Audit log

#### Steps to test this PR:
1. Repeat and this time the audit log should have entries
2. Run the automated tests

### CODE REVIEW

**Comment by dbhurley** [comment]
The problem attempting to be solved here is listed as follows: 
> Contacts and companies created by Pipedrive's webhook (import) did not dispatch events so plugins and other listeners were not notified of changes made.

However the issue is not _properly addressed_ by adding in plugin-specific code to the generic CRMBundle Event Listeners. Instead, the offending plugin-in (in this case Pipedrive) should be fixed. 

@npracht and @thecodeine - would you please look at the original code and comment on this problem? We should identify a solution that does not involve "hacking" the core, non-plugin specific files to make exception for a single plugin.

File: plugins/MauticCrmBundle/EventListener/CompanySubscriber.php [hunk 1]
```diff
@@ -60,28 +60,42 @@ public static function getSubscribedEvents()
      */
     public function onCompanyPostSave(Events\CompanyEvent $event)
     {
+        $company = $event->getCompany();
+        if ($company->getEventData('pipedrive.webhook')) {
```
Comment (dbhurley) [inline plugins/MauticCrmBundle/EventListener/CompanySubscriber.php]: This plugin-specific code should not be in the generic EventListener.

File: plugins/MauticCrmBundle/EventListener/CompanySubscriber.php [hunk 2]
```diff
@@ -60,28 +60,42 @@ public static function getSubscribedEvents()
...
+            // Don't export what was just imported
+            return;
+        }
+
+        /** @var PipedriveIntegration $integrationObject */
         $integrationObject = $this->integrationHelper->getIntegrationObject(PipedriveIntegration::INTEGRATION_NAME);
```
Comment (dbhurley) [inline plugins/MauticCrmBundle/EventListener/CompanySubscriber.php]: @npracht @thecodeine and @Webmecanik - this and all other instances where "Pipedrive" is referred to in a generic CRMBundle file should be removed.

File: plugins/MauticCrmBundle/EventListener/CompanySubscriber.php [hunk 3]
```diff
@@ -60,28 +60,42 @@ public static function getSubscribedEvents()
...
-        $this->companyExport->pushCompany($event->getCompany());
+        $this->companyExport->pushCompany($company);
     }
 
     /**
      * @param Events\LeadEvent $event
      */
     public function onCompanyPreDelete(Events\CompanyEvent $event)
     {
+        $company = $event->getCompany();
```
Comment (dbhurley) [inline plugins/MauticCrmBundle/EventListener/CompanySubscriber.php]: Same issue here.

File: plugins/MauticCrmBundle/EventListener/LeadSubscriber.php [hunk 4]
```diff
@@ -61,43 +61,64 @@ public static function getSubscribedEvents()
      */
     public function onLeadPostSave(Events\LeadEvent $event)
     {
+        $lead = $event->getLead();
+        if ($lead->getEventData('pipedrive.webhook')) {
```
Comment (dbhurley) [inline plugins/MauticCrmBundle/EventListener/LeadSubscriber.php]: Same issue here. Pipedrive specfic code should not be included in generic `EventListener/LeadSubscriber.php`

**Comment by npracht** [comment]
Hello @dbhurley !
We didn't develop it ourselves unfortunately i don't know how it is done. @thecodeine is at the origin from it but i don't have news from them..
@kuzmany has fixed some issues and developed some extra features on it. Zdeno, maybe you can help ?

**Comment by javjim-mautic** [comment]
Can someone test this too? I am getting duplicates for each contact

**Comment by javjim-mautic** [comment]
Was able to test it and its working properly

**Comment by dongilbert** [comment]
+1 works as expected. Thanks @alanhartless

**Pull request merged**